include/blmc_drivers/devices/spi_bus.hpp

```diff
@@ -0,0 +1,282 @@
+     * Inherited from MotorBoardInterface
+     */
+    virtual std::shared_ptr<const MotorInterface::ScalarTimeseries>
+    get_sent_control(
+        const size_t udriver_id,
+        const MotorBoardInterface::ControlIndex& index) const;
+
+    /**
+     * @copydoc MotorBoardInterface::get_sent_command()
+     * Inherited from MotorBoardInterface
+     */
+    virtual std::shared_ptr<const MotorBoardInterface::CommandTimeseries>
+    get_sent_command(const size_t udriver_id) const;
+
+    /**
+     * Setters
+     */
+    
+    /**
+     * @copydoc MotorBoardInterface::set_control()
+     * Inherited from MotorBoardInterface
+     */
+    virtual void set_control(const size_t udriver_id, const double& control,
+                             const MotorBoardInterface::ControlIndex& index);
+    
+    /**
+     * @copydoc MotorBoardInterface::set_command()
+     * Inherited from MotorBoardInterface
+     */
+    virtual void set_command(const size_t udriver_id, 
+                             const MotorBoardCommand& command);
+
+    /**
+     * @copydoc MotorBoardInterface::send_if_input_changed()
+     * Inherited from MotorBoardInterface. This particualr instance does not
+     * actually check if it is is a new command or control as the full status of
+     * the robot is exchange at every tick.
+     */
+    virtual void send_if_input_changed();
+
+    /**
+     * @brief return s only once board and motors are ready.
+     */
+    bool is_ready();
+
+    /**
+     * @brief Wait until the robot is ready.
+     */
+    void wait_until_ready();
+
+private:
+    /**
+     * Private methods
+     */
+
+    /**
+     * @brief This is the helper function used for spawning the real time
+     * thread.
+     *
+     * @param instance_pointer is the current object in this case.
+     * @return THREAD_FUNCTION_RETURN_TYPE depends on the current OS.
+     */
+    static THREAD_FUNCTION_RETURN_TYPE loop(void* instance_pointer)
+    {
+        static_cast<SpiBus*>(instance_pointer)->loop();
+        return THREAD_FUNCTION_RETURN_VALUE;
+    }
+
+    /**
+     * @brief This is the real time thread that streams the data to/from the
+     * main board. 
+     */
+    void loop();
+
+    /**
+     * @brief Send the newest control stored in the time series.
+     */
+    void send_newest_command();
+
+    /**
+     * @brief Send the newest control stored in the time series.
+     */
+    void send_newest_controls();
+
+    /**
+     * Communication related attributes
+     */
+
+    /**
+     * @brief Main board interface sdk:
+     * https://github.com/open-dynamic-robot-initiative/master-board
+     */
+    std::shared_ptr<MasterBoardInterface> main_board_interface_;
+
+    /**
+     * @brief nb_udrivers_ is the number of the udriver controlled by the
+     * main board.
+     */
+    size_t nb_udrivers_;
+
+    /**
+     * Outputs
+     */
+
+    /**
+     * @brief measurement_ contains all the measurements acquiered from the CAN
```

Comment:
No need to mention the variable name in the doc
```suggestion
     * @brief All the measurements acquiered from the CAN
```

Reply:
Also the `This is the` in the docs below could be removed to have more compact documentation.